src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts

```diff
@@ -4,52 +4,152 @@
-import { ExtensionManagementService } from 'vs/workbench/services/extensionManagement/common/extensionManagementService';
+import { MockContextKeyService } from 'vs/platform/keybinding/test/common/mockKeybindingService';
 import { NullLogService } from 'vs/platform/log/common/log';
-import { NBTestQueryManagementService } from 'sql/workbench/contrib/notebook/test/nbTestQueryManagementService';
-import { INotebookService } from 'sql/workbench/services/notebook/browser/notebookService';
-import { Emitter } from 'vs/base/common/event';
 import { Registry } from 'vs/platform/registry/common/platform';
-import { NotebookProviderRegistration, INotebookProviderRegistry, Extensions } from 'sql/workbench/services/notebook/common/notebookRegistry';
-import { MockContextKeyService } from 'vs/platform/keybinding/test/common/mockKeybindingService';
+import { IWorkspaceStorageChangeEvent, StorageScope } from 'vs/platform/storage/common/storage';
+import { ExtensionManagementService } from 'vs/workbench/services/extensionManagement/common/extensionManagementService';
+import { TestFileService, TestLifecycleService } from 'vs/workbench/test/browser/workbenchTestServices';
+import { TestExtensionService, TestStorageService } from 'vs/workbench/test/common/workbenchTestServices';
+
+/**
+ * class to mock azdata.nb.ServerManager object
+ */
+class TestServerManager implements azdata.nb.ServerManager {
+	isStarted: boolean = true; //by default our mock creates ServerManager in started mode.
+	onServerStarted: Event<void>;
+	async startServer(kernelSpec: azdata.nb.IKernelSpec): Promise<void> {
+		this.isStarted = true;
+	}
+	async stopServer(): Promise<void> {
+		this.isStarted = false;
+	}
+
+}
+
+/**
+ * TestNotebookManager - creates a NotebookManager object that helps keep track of state needed by testing
+ */
+class TestNotebookManager implements INotebookManager {
+	contentManager: undefined;
+	sessionManager: undefined;
+	constructor(
+		public providerId: string = 'providerId1',
+		public serverManager: TestServerManager = new TestServerManager()
+	) { }
+}
+
+/**
+ * * TestNotebookProvider - creates a NotebookManager object that helps keep track of state needed by testing
+ */
+class TestNotebookProvider implements INotebookProvider {
+	constructor(
+		public providerId: string = 'providerId1',
+		public manager: TestNotebookManager = new TestNotebookManager(providerId)
+	) { }
+
+	getNotebookManager(uri: URI): Thenable<INotebookManager> {
+		return Promise.resolve(this.manager);
+	}
 
-suite('Notebook Service Tests', function (): void {
-	let notebookService: INotebookService;
+	handleNotebookClosed(_notebookUri: URI): void {
+		// do nothing
+	}
+}
 
-	let installEvent: Emitter<InstallExtensionEvent>,
-		didInstallEvent: Emitter<DidInstallExtensionEvent>,
-		uninstallEvent: Emitter<IExtensionIdentifier>,
-		didUninstallEvent: Emitter<DidUninstallExtensionEvent>;
+suite('Test class ProviderDescriptor:', () => {
```

Comment:
You don't need to prefix every suite with `Test class` - we know this is a test already. Just the name of the thing being tested is fine for the top level suite.

Reply:
Fixed